### PR TITLE
migrate haproxy config tool/script to master branch

### PR DESCRIPTION
**Pull request by chenqianfzh**
This PR migrates the HaProxy config tool/script from PoC branch to master.

It is verified in one-box.

Verfication with kubemark clusters will be done after the kubemakr scale-out change is also migrated to master.

### CODE REVIEW

File: cmd/haproxy-cfg-generator/cfg_generator.go [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+		os.Exit(1)
+	}
+
+	if net.ParseIP(resourcePartitionIP) == nil {
+		LogError("\nIP address (%v) is invalid.\n", resourcePartitionIP)
+		os.Exit(1)
+	}
+}
+
+func get_tp_request_acl() string {
```
Comment (pdgetrf) [inline cmd/haproxy-cfg-generator/cfg_generator.go:98]: some functions in this file with "fancy" logic might need unit tests

Comment (chenqianfzh) [reply cmd/haproxy-cfg-generator/cfg_generator.go]: As discussed, this tool is now test-protected.

Also the "patch haproxy for prometheus" will be activated only when a specific env var is "true".

**Pull request closed without merging**